service/src/main/java/com/wso2telco/dep/tpservice/pool/TokenReGenarator.java

```diff
@@ -98,21 +112,55 @@ public TokenDTO reGenarate(final WhoDTO who, final TokenDTO oldToken) throws Tok
 		// validate response message
 		if (Strtoken != null && Strtoken.length() > 0) {
 
-			JSONObject jsontoken = new JSONObject(Strtoken);
+			try {
+				jsontoken = new JSONObject(Strtoken);
```

Comment:
When IS is down API manager send a soap request to the token pool. So this can't be converted to Json.  So we need to retry in that scenarios.